--- sqlgen-0.3.0/src/sqlgen/postgres/to_sql.cpp	2025-10-22 16:21:09.527684431 +0100
+++ sqlgen-0.3.0/src/sqlgen/postgres/to_sql.cpp	2025-10-22 16:20:59.935751189 +0100
@@ -751,6 +751,9 @@
   return _table_or_query.visit([](const auto& _t) -> std::string {
     using Type = std::remove_cvref_t<decltype(_t)>;
     if constexpr (std::is_same_v<Type, dynamic::Table>) {
+      if (_t.schema) {
+        return wrap_in_quotes(*_t.schema) + "." + wrap_in_quotes(_t.name);
+      }
       return wrap_in_quotes(_t.name);
     } else {
       return "(" + select_from_to_sql(*_t) + ")";
